gatsby-node.js

```diff
@@ -84,55 +84,80 @@ exports.onCreateBabelConfig = ({actions}) => {
 	});
  };
 
-function createSlug ({relativePath}) {
+function createSlug ({absolutePath, relativePath}) {
 	let slug;
 	const parsedFilePath = path.parse(relativePath);
-	if (parsedFilePath.name !== 'index' && parsedFilePath.dir !== '') {
-		slug = `/${parsedFilePath.dir}/${parsedFilePath.name}/`;
-	} else if (parsedFilePath.dir === '') {
-		slug = `/${parsedFilePath.name}/`;
-	} else {
+	const paredAbsoluteFilePath = path.parse(absolutePath);
+
+	if(paredAbsoluteFilePath.dir.indexOf('jsdocs')) {
 		slug = `/${parsedFilePath.dir}/`;
+	} else {
+		if (parsedFilePath.name !== 'index' && parsedFilePath.dir !== '') {
+			slug = `/${parsedFilePath.dir}/${parsedFilePath.name}/`;
+		} else if (parsedFilePath.dir === '') {
+			slug = `/${parsedFilePath.name}/`;
+		} else {
+			slug = `/${parsedFilePath.dir}/`;
+		}
 	}
+
 	return slug;
 }
 
-async function onCreateNode ({node, actions, getNode, loadNodeContent}) {
+async function onCreateNode ({node, actions, getNode}) {
 	const {createNodeField, createNode, createParentChildLink} = actions;
 	let slug;
 	if (node.internal.type === 'MarkdownRemark') {
 		const fileNode = getNode(node.parent);
-		slug = createSlug(fileNode);
+		if(fileNode.internal.type === 'File') {
+			slug = createSlug(fileNode);
 
+			// Add slug as a field on the node.
+			createNodeField({node, name: 'slug', value: slug});
```

Comment:
As I checked, if fileNode.internal.type is 'File', createNodeField will be called twice (in L116 and L121). Is it designed to called twice?

Reply:
It's fixed.